src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -83,6 +83,26 @@ internal Binder Binder
             return new ForEachLocal(containingMethod, binder, typeSyntax, identifierToken, collection, LocalDeclarationKind.ForEachIterationVariable);
         }
 
+        public static SourceLocalSymbol MakeDeconstructionLocal(
+            Symbol containingSymbol,
+            Binder binder,
+            TypeSyntax typeSyntax,
+            SyntaxToken identifierToken,
+            TypeSyntax parentTypeSyntax)
+        {
+            Debug.Assert(parentTypeSyntax == null || parentTypeSyntax.IsVar);
+
+            if (parentTypeSyntax != null || typeSyntax.IsVar)
```

Comment:
We are still dealing with two type syntaxes here. I think this function should take just one. The caller should decide which node to pass to this function, but it should be one node. We are always passing typeSyntax to the local symbol. Could it be null? In cases when it is not null and not var, is there a reason to create PossiblyImplicitlyTypedDeconstructionLocalSymbol?